src/crystal/system/win32/file_descriptor.cr

```diff
@@ -162,4 +163,22 @@ module Crystal::System::FileDescriptor
 
     bytes_read
   end
+
+  def self.from_stdio(fd)
+    console_handle = false
+    handle = LibC._get_osfhandle(fd)
+    if handle != -1
+      if LibC.GetConsoleMode(LibC::HANDLE.new(handle), out _) != 0
+        console_handle = true
+      end
+    end
+
+    io = IO::FileDescriptor.new(fd)
+    if console_handle
+      io.sync = true
```

Comment:
Wait, isn't this inverted? console should be flush_on_newline, otherwise sync

Reply:
https://github.com/crystal-lang/crystal/blob/0.34.0/src/kernel.cr#L15-L20
```
# - if it's a TTY device (like the console) then `sync` is `true`,
#   meaning that output will be outputted as soon as it is written
#   to STDOUT. This is so users can see real time output data.
# - if it's not a TTY device (like a file, or because the output
#   was piped to a file) then `sync` is `false` but `flush_on_newline`
#   is `true`. This is so that if you pipe the output to a file, and,
```

Reply:
Maybe you could add a reference to this as a doc comment like in the unix implementation?